src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesAEAD.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal partial class AesAEAD
+    {
+        public static void CheckKeySize(int keySizeInBits)
+        {
+            if (keySizeInBits != 128 && keySizeInBits != 192 && keySizeInBits != 256)
+            {
+                throw new CryptographicException(SR.Cryptography_InvalidKeySize);
+            }
+        }
+
+        public static bool MatchesKeySizes(int size, KeySizes keySizes)
+        {
+            if (size < keySizes.MinSize || size > keySizes.MaxSize)
+                return false;
+
+            return ((size - keySizes.MinSize) % keySizes.SkipSize) == 0;
+        }
+
+        public static void CheckArgumentsForNull(
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> plaintext,
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> tag)
+        {
+            if (nonce == null)
+                throw new ArgumentNullException(nameof(nonce));
```

Comment:
Why are we still comparing spans to null?

Reply:
It looks like the only callers of these are the ones using arrays.  I'd just lift the checks into the 4 methods that use them, so it's more clear that argument validation is happening.

Reply:
I'll change the signature to `byte[]` - I prefer to have one common place in case we change our mind how to format the exception